### PR TITLE
fix: put checkout_cores test behind a single-threaded feature

### PR DESCRIPTION
**Pull request by cryptonemo**
feat: add checkout_cores test to CI, run single threaded
doc: fix typo

A possible way to address #1378

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -147,6 +147,7 @@ jobs:
             ulimit -u 20000
             ulimit -n 20000
             cargo +<< pipeline.parameters.nightly-toolchain >> -Zpackage-features test --all --verbose --release lifecycle -- --ignored --nocapture
+            cargo +<< pipeline.parameters.nightly-toolchain >> -Zpackage-features test --features single-threaded --release checkout_cores -- --test-threads=1
```
Comment (dignifiedquire) [inline .circleci/config.yml:150]: this is missing the `--all` package specificer

Comment (cryptonemo) [reply .circleci/config.yml:150]: Correct, that's the purpose of this PR.  The all is performed, followed by the single checkout_cores test that is isolated.

Comment (dignifiedquire) [reply .circleci/config.yml:150]: but it still needs to select a package?

Comment (cryptonemo) [reply .circleci/config.yml:150]: The CI log shows what I see locally, which is the single test running (after the all): https://app.circleci.com/pipelines/github/filecoin-project/rust-fil-proofs/2838/workflows/c44d3b1c-e69e-43cc-8c06-6c4e533ccf92/jobs/62773

Comment (cryptonemo) [reply .circleci/config.yml:150]: Ok, adding the package is more direct, I'll change that (just tried locally).

**Comment by cryptonemo** [comment]
Currently, the sdr checkout test is done on the pairing side and not mirrored on the blst side.  I don't think there's any need to run it more than once.  Should it be moved out to a separate job entirely? I only really want to know if it's failing for some reason on CI.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**